src/Microsoft.VisualStudio.Services.Agent/SecretMask.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    public interface ISecretMask
```

Comment:
I personally don't like extension methods unless there is a really good reason. It just makes things more confusing to figure out where the code lives. I don't understand what benefit we get from making these extension methods rather than methods on the SecretMasker class.

We don't have 100 mask classes and never will.